### PR TITLE
Bump vimeo/psalm from 4.4.1 to 4.6.3

### PR DESCRIPTION
**Pull request by malukenho**
Favor this PR over https://github.com/Ocramius/ProxyManager/pull/674

### CODE REVIEW

File: tests/ProxyManagerTest/Functional/AccessInterceptorScopeLocalizerFunctionalTest.php [hunk 1]
```diff
@@ -290,7 +292,8 @@ public function testCanWriteToArrayKeysInPublicProperty(): void
     public function testWillNotModifyRetrievedPublicProperties(): void
     {
         $instance = new ClassWithPublicProperties();
-        $proxy    = (new AccessInterceptorScopeLocalizerFactory())->createProxy($instance);
+        /** @psalm-var AccessInterceptorInterface<object>&ClassWithPublicProperties $proxy */
```
Comment (Ocramius) [inline tests/ProxyManagerTest/Functional/AccessInterceptorScopeLocalizerFunctionalTest.php:295]: `<object>` should be `<ClassWithPublicProperties>`

File: tests/ProxyManagerTest/Functional/AccessInterceptorScopeLocalizerFunctionalTest.php [hunk 2]
```diff
@@ -311,7 +314,8 @@ public function testWillNotModifyRetrievedPublicProperties(): void
     public function testWillModifyByRefRetrievedPublicProperties(): void
     {
         $instance = new ClassWithPublicProperties();
-        $proxy    = (new AccessInterceptorScopeLocalizerFactory())->createProxy($instance);
+        /** @psalm-var AccessInterceptorInterface<object>&ClassWithPublicProperties $proxy */
```
Comment (Ocramius) [inline tests/ProxyManagerTest/Functional/AccessInterceptorScopeLocalizerFunctionalTest.php:317]: `<object>` should be `<ClassWithPublicProperties>`

File: tests/ProxyManagerTest/Functional/AccessInterceptorScopeLocalizerFunctionalTest.php [hunk 3]
```diff
@@ -271,7 +272,8 @@ public function testPropertyUnset(object $instance, AccessInterceptorInterface $
     public function testCanWriteToArrayKeysInPublicProperty(): void
     {
         $instance = new ClassWithPublicArrayPropertyAccessibleViaMethod();
-        $proxy    = (new AccessInterceptorScopeLocalizerFactory())->createProxy($instance);
+        /** @psalm-var AccessInterceptorInterface<object>&ClassWithPublicArrayPropertyAccessibleViaMethod $proxy */
```
Comment (Ocramius) [inline tests/ProxyManagerTest/Functional/AccessInterceptorScopeLocalizerFunctionalTest.php:275]: `<object>` should be `<ClassWithPublicArrayPropertyAccessibleViaMethod>`

File: tests/ProxyManagerTest/Functional/AccessInterceptorScopeLocalizerFunctionalTest.php [hunk 4]
```diff
@@ -180,6 +180,7 @@ public function testMethodCallsAfterUnSerialization(
         array $params,
         $expectedValue
     ): void {
+        /** @psalm-var AccessInterceptorInterface<object> $proxy */
         $proxy = unserialize(serialize((new AccessInterceptorScopeLocalizerFactory())->createProxy($instance)));
         assert($proxy instanceof AccessInterceptorInterface);
```
Comment (Ocramius) [inline tests/ProxyManagerTest/Functional/AccessInterceptorScopeLocalizerFunctionalTest.php:185]: These assertions become redundant now, no?

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by malukenho** [comment]
I've tried your suggestions before — and tried it again right now. why not? — it didn't work 😬

**Comment by Ocramius** [comment]
> it didn't work :grimacing: 

What didn't work?

**Comment by malukenho** [comment]
All of your review comments.

**Comment by Ocramius** [comment]
Yes, but what didn't work? :D

**Comment by malukenho** [comment]
The changes on the psalm comments you suggested. It basically gave me the similar errors to the #674

**Pull request closed without merging**